packages/helpers/src/dkim/index.ts

```diff
@@ -22,20 +28,28 @@ export async function verifyDKIMSignature(
 ): Promise<DKIMVerificationResult> {
   let dkimResult = await tryVerifyDKIM(email, domain);
 
-  if (dkimResult.status.result !== "pass" && tryRevertARCChanges) {
-    console.info("DKIM verification failed. Trying to verify after reverting forwarder changes...");
-
-    const modified = await revertForwarderChanges(email.toString());
+  // If DKIM verification fails, revert common modifications made by ARC and try again.
+  if (dkimResult.status.comment === "bad signature" && tryRevertARCChanges) {
```

Comment:
before we do this, we should add the zkp2p code where they try to process the email via doing the tabs -> spaces replacement as well as removing a [LABEL] tag from an email. i think we shuold try as hard as possible to get a dkim before moving to arc, and output that dkim failed and we are moving to arc before doing modifications and claiming dkim passes.

Reply:
> tabs -> spaces replacement as well as removing a [LABEL] tag from an email
Do you know where this is implemented? I think those changes are also part of ARC? We should definitely be adding other ARC changes.

> output that dkim failed and we are moving to arc
The `revertCommonARCModifications` will print a log when modifications are done, or if no changes are done.

Reply:
> > tabs -> spaces replacement as well as removing a [LABEL] tag from an email
> 
> Do you know where this is implemented? I think those changes are also part of ARC? We should definitely be adding other ARC changes.
> 
> > output that dkim failed and we are moving to arc
> 
> The `revertCommonARCModifications` will print a log when modifications are done, or if no changes are done.

Check out the zkp2p fixes -- basically like if you revert those changes, then you can get DKIM to pass, not just ARC. I think they noticed this on some Venmo emails and diffed them.

Reply:
We are doing the same approach - right now only Message-Id is reverted, but can add removing labels.

https://github.com/zkp2p/zk-p2p/blob/develop/client/src/components/ProofGen/validation/hdfc.tsx#L85 seems to be very specific to hdfc pasted emails.

I think we can merge this and address 162, 165 and other sanitizations in another PR.